File: bootstrap/src/main/java/net/dontdrinkandroot/wicket/bootstrap/component/button/IButton.java
package net.dontdrinkandroot.wicket.bootstrap.component.button;

import org.apache.wicket.model.IModel;
import org.apache.wicket.util.io.IClusterable;

import net.dontdrinkandroot.wicket.bootstrap.css.ButtonSize;
import net.dontdrinkandroot.wicket.bootstrap.css.ButtonStyle;


public interface IButton
{

	ButtonSize getButtonSize();

	IClusterable setButtonSize(ButtonSize buttonSize);

	IClusterable setButtonSizeModel(IModel<ButtonSize> buttonSizeModel);

	ButtonStyle getButtonStyle();

	IClusterable setButtonStyle(ButtonStyle buttonStyle);

	IClusterable setButtonStyleModel(IModel<ButtonStyle> buttonStyleModel);

}

File: bootstrap/src/main/kotlin/net/dontdrinkandroot/wicket/bootstrap/component/button/IButton.kt
package net.dontdrinkandroot.wicket.bootstrap.component.button

import net.dontdrinkandroot.wicket.bootstrap.css.ButtonSize
import net.dontdrinkandroot.wicket.bootstrap.css.ButtonStyle
import org.apache.wicket.model.IModel
import org.apache.wicket.util.io.IClusterable

interface IButton {

    fun getButtonSize(): ButtonSize?
    fun setButtonSize(buttonSize: ButtonSize?): IClusterable

    @Deprecated("Model will usually be bound already")
    fun setButtonSizeModel(buttonSizeModel: IModel<ButtonSize?>): IClusterable

    fun getButtonStyle(): ButtonStyle
    fun setButtonStyle(buttonStyle: ButtonStyle): IClusterable

    @Deprecated("Model will usually be bound already")
    fun setButtonStyleModel(buttonStyleModel: IModel<ButtonStyle>): IClusterable
}